bsmetadata/preprocessing_tools/wikipedia_desc_utils.py

```diff
@@ -45,14 +45,13 @@ def fetch_website_description_from_keyword(self, keyword: str) -> Optional:
         return self.cache[keyword]
 
     def fetch_entity_description_from_keyword(self, keyword: str) -> str:
+
         title = string.capwords(keyword)
-        text = self.wiki_dump_db.get_paragraphs(title)[0].text
-        text = re.sub(r"\((?:[^)(]|\([^)(]*\))*\)", "", text)
-        text = nltk.sent_tokenize(text)[0]
+        text = self.fetch_wikipedia_description_for_title(title)
 
         if text is None:
-            text = self.wiki_dump_db.get_paragraphs(self.redirects_map[keyword])[0].text
-            text = nltk.tokenize.sent_tokenize(text)[0]
+            title = self.redirects_map[keyword]
```

Comment:
Sorry if my previous comment was not very clear but I think this can throw a key error if the key is not found. We can wither handle it using try catch or keep a prior check ` if key in redirects_map:`

Reply:
Oh yes, we should indeed tackle this issue if the keyword has a probability to not be in `redirects_map`.

I am in favor of your second suggestion @shanyas10 (the use of try/catch need to be avoided when we can IMO) :

```python
if text is None and keyword in self.redirects_map:
    title = self.redirects_map[keyword]
    text = self.fetch_wikipedia_description_for_title(title)
```

Reply:
Updated.

Reply:
Hope the main issue, ie. batching is fixed now. :)